core/templates/pages/story-editor-page/navbar/story-editor-navbar.component.ts

```diff
@@ -96,40 +96,44 @@ export class StoryEditorNavbarComponent implements OnInit {
 
   discardChanges(): void {
     this.undoRedoService.clearChanges();
-    this.storyEditorStateService.loadStory(this.story.getId());
+    if (this.story !== undefined) {
+      this.storyEditorStateService.loadStory(this.story.getId());
+    }
     this._validateStory();
     this.forceValidateExplorations = true;
   }
 
   private _validateStory(): void {
-    this.validationIssues = this.story.validate();
-    let nodes = this.story.getStoryContents().getNodes();
-    let skillIdsInTopic = (
-      this.storyEditorStateService.getSkillSummaries().map(
-        skill => skill.id));
-    if (this.validationIssues.length === 0 && nodes.length > 0) {
-      let prerequisiteSkillValidationIssues = (
-        this.storyValidationService
-          .validatePrerequisiteSkillsInStoryContents(
-            skillIdsInTopic, this.story.getStoryContents()));
-      this.validationIssues = (
-        this.validationIssues.concat(
-          prerequisiteSkillValidationIssues));
-    }
-    if (this.storyEditorStateService.getStoryWithUrlFragmentExists()) {
-      this.validationIssues.push(
-        'Story URL fragment already exists.');
+    if (this.story !== undefined) {
+      this.validationIssues = this.story.validate();
```

Comment:
Could you rather do?
```suggestion
    if (this.story == undefined) {
      return;
    }
    this.validationIssues = this.story.validate();
```

Also, should we fire an error if `story` is `undefined`?

Reply:
Done.